File: src/core/validators/CommonSimpleType/CommonStringMinLengthMustNotBeGreaterThanMaxLength.ts
import { ValidationRuleBase } from "../ValidationRuleBase";
export class CommonStringMinLengthMustNotBeGreaterThanMaxLength extends ValidationRuleBase<MetaEdGrammar.CommonStringContext>
{
    public isValid(context: MetaEdGrammar.CommonStringContext): boolean {
        if (context.minLength() == null)
            return true;
        let minLength = Number(context.minLength().MinLength());
        let maxLength = Number(context.maxLength().MaxLength());
        return minLength <= maxLength;
    }
    public getFailureMessage(context: MetaEdGrammar.CommonStringContext): string {
        return `Common String '${context.commonStringName().GetText()}' has min length greater than max length.`;
    }
}

File: src/core/validators/CommonSimpleType/CommonStringMinLengthMustNotBeGreaterThanMaxLength.cs
using System;
using MetaEd.Grammar.Antlr;

namespace MetaEd.Core.Validator.CommonSimpleType
{
    public class CommonStringMinLengthMustNotBeGreaterThanMaxLength : ValidationRuleBase<MetaEdGrammar.CommonStringContext>
    {
        public override bool IsValid(MetaEdGrammar.CommonStringContext context)
        {
            // minLength is optional
            if (context.minLength() == null) return true;

            // if there are convert exceptions, let it bomb out -- language parser should have handled
            var minLength = Convert.ToInt32(context.minLength().MinLength());
            var maxLength = Convert.ToInt32(context.maxLength().MaxLength());

            return minLength <= maxLength;
        }

        public override string GetFailureMessage(MetaEdGrammar.CommonStringContext context)
        {
            return string.Format("Common String '{0}' has min length greater than max length.", context.commonStringName().GetText());
        }
    }
}
